scripts/coverage.sh

```diff
@@ -29,15 +29,24 @@ export CARGO_INCREMENTAL=0
 export RUST_BACKTRACE=1
 export RUST_MIN_STACK=8388608
 
-echo "--- remove old coverage results"
+echo "--- revert target dir to pre-run state to remove old coverage results"
+# Remove me after awhile since we transition to new cache pruning method below
 if [[ -d target/cov ]]; then
   find target/cov -name \*.gcda -print0 | xargs -0 rm -f
 fi
 rm -rf target/cov/$reportName
 
+if [[ -e target/cov/build-finished ]]; then
+  echo "Reverting files"
+  find target/cov -newer target/cov/build-finished -type f -print -delete
+  echo "Reverting (empty) directories"
+  find target/cov -newer target/cov/build-finished -type d -empty -print -delete
```

Comment:
After quick googling, I bet `-delete` and `-newer` is available on macOS. Dunno why the current code uses print0-and-xargs trick needlessly.

Reply:
As a matter of fact, this can't be replaced with the trick because we want to recursively remove directories which got empty. The trick is effectively a BFS, and this `find` is effectively a DFS.

Reply:
Confirmed macOS `find` has `-delete` and `-newer`